climada/engine/supplychain.py

```diff
@@ -0,0 +1,342 @@
+                    table_info_pos={'range_rows':(5, 2469),
+                                    'range_cols':(4, 2468),
+                                    'col_iso3':2,'col_sectors':1}):
+        """Read multi-regional input-output tables of the 2016 release of the
+        WIOD project: http://www.wiod.org/database/wiots16
+
+        Parameters
+        ----------
+        year : int
+            Year of WIOD table to use. Valid years go from 2000 to 2014.
+            Default year is 2014.
+        table_info_pos: dict
+            Dictionary with info about how to properly read the WIOD table.
+            Default values allow reading the full WIOD table. Keys are:
+                range_rows : tuple
+                    initial and end positions of data along rows.
+                range_cols : tuple
+                    initial and end positions of data along columns.
+                col_iso3 : int
+                    column with countries names in ISO3 codes.
+                col_sectors : int
+                    column with sector names.
+        References
+        ----------
+        [1] Timmer, M. P., Dietzenbacher, E., Los, B., Stehrer, R. and de Vries, G. J.
+        (2015), "An Illustrated User Guide to the World Input–Output Database: the Case
+        of Global Automotive Production", Review of International Economics., 23: 575–605
+
+        """
+
+        file_name = 'WIOT{}_Nov16_ROW.xlsb'.format(year)
+        file_loc = WIOD_DIRECTORY / file_name
+
+        if not file_loc in WIOD_DIRECTORY.iterdir():
+            download_link = WIOD_FILE_LINK + file_name
+            u_fh.download_file(download_link, download_dir=WIOD_DIRECTORY)
+            LOGGER.info('Downloading WIOD table for year %s', year)
+        mriot = pd.read_excel(file_loc, engine='pyxlsb')
+
+        col_iso3 = table_info_pos['col_iso3']
+        col_sectors = table_info_pos['col_sectors']
+        start_row, end_row = table_info_pos['range_rows']
+        start_col, end_col = table_info_pos['range_cols']
+
+        self.sectors = mriot.iloc[start_row:end_row, col_sectors].unique()
+        self.mriot_reg_names = mriot.iloc[start_row:end_row, col_iso3].unique()
+        self.mriot_data = mriot.iloc[start_row:end_row,
+                                     start_col:end_col].values
+        self.total_prod = mriot.iloc[start_row:end_row, -1].values
+        self.reg_pos = {
+            name: range(len(self.sectors)*i, len(self.sectors)*(i+1))
+            for i, name in enumerate(self.mriot_reg_names)
+            }
+        self.mriot_type = 'WIOD'
+
+    def calc_sector_direct_impact(self, hazard, exposure, imp_fun_set,
+                                  selected_subsec="service"):
+        """Calculate direct impacts.
+
+        Parameters:
+        ----------
+        hazard : Hazard
+            Hazard object for impact calculation.
+        exposure : Exposures
+            Exposures object for impact calculation.
```

Comment:
Add that when using the WIOD data, the `reg_id` of the exposure must conform to ... format.

Reply:
But that is trivial, isn't it? If you use a table with country-wide exposures, then you use countries' exposures; and in CLIMADA countries exposure are supported by countries_by_numeric (see also other comment).

Reply:
Countries exposure are not necessarily supported by `countries_by_numeric`. Only exposures that use the ISO3 standard do so. This is no big deal. It just must be documented.